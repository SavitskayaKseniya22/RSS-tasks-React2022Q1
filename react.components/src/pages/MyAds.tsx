import React from 'react';
import { AddAdv } from '../components/AddMyAdvForm';

export class MyAds extends React.Component {
  constructor(props: string) {
    super(props);
  }
  render() {
    return (
      <div data-testid="about-us">
        <h1>my advertisements</h1>
        <AddAdv />
      </div>
    );
  }
}
