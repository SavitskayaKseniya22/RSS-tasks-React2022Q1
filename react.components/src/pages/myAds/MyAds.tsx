import { useState } from 'react';
import { AdsList } from '../../components/adsList/AdsList';
import { Form } from '../../components/form/Form';
import { CardProps, FormStateTypes } from '../../interfaces';
import './myAds.css';

export function MyAds() {
  const [state, setState] = useState<FormStateTypes>({ savedCards: [] });

  const changeState = (object: CardProps) => {
    setState({
      ...state,
      savedCards: [...state.savedCards, object],
    });
  };

  return (
    <div data-testid="my-ads" className="my-ads">
      <h1>my advertisements</h1>
      <Form changeState={changeState} />
      {state.savedCards.length > 0 && <AdsList savedCards={state.savedCards} />}
    </div>
  );
}
