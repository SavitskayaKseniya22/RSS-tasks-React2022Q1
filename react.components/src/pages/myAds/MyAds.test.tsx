import { render, screen } from '@testing-library/react';
import { MyAds } from './MyAds';

test('check MyAds appearance', () => {
  render(<MyAds />);
  expect(screen.getByTestId('my-ads')).toBeInTheDocument();
  expect(screen.getByText('my advertisements')).toBeInTheDocument();
  expect(screen.getByTestId('form-ad')).toBeInTheDocument();
});
