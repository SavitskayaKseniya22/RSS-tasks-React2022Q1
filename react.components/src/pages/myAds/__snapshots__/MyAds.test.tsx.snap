// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check form appearance 1`] = `
<div
  class="my-ads"
  data-testid="my-ads"
>
  <h1>
    my advertisements
  </h1>
  <div
    class="advertisements"
    data-testid="advertisements"
  >
    <form
      class="add-adv"
      data-testid="form-ad"
      novalidate=""
    >
      <label
        class="big-field"
      >
        <h3>
          Title:
        </h3>
        <input
          data-testid="form__title"
          name="title"
          placeholder="Please enter a valid title. String must contain at least 6 characters."
          type="text"
        />
        <span
          class="error-note"
          data-testid="form__title-error"
        />
      </label>
      <label
        class="big-field"
      >
        <h3>
          Description:
        </h3>
        <input
          data-testid="form__description"
          name="description"
          placeholder="Please enter a valid description. String must contain at least 10 characters."
          type="text"
        />
        <span
          class="error-note"
          data-testid="form__description-error"
        />
      </label>
      <label
        class="middle-field"
      >
        <h3>
          Tel:
        </h3>
        <input
          data-testid="form__tel"
          name="tel"
          placeholder="Please enter a valid number. Number must contain at least 5 digits."
          type="tel"
        />
        <span
          class="error-note"
          data-testid="form__tel-error"
        />
      </label>
      <label
        class="middle-field"
      >
        <h3>
          E-mail:
        </h3>
        <input
          data-testid="form__email"
          name="email"
          placeholder="Please enter a valid email"
          type="email"
        />
        <span
          class="error-note"
          data-testid="form__email-error"
        />
      </label>
      <label
        class="small-field"
      >
        <h3>
          Area (㎡):
        </h3>
        <input
          data-testid="form__area"
          name="area"
          placeholder="Enter the area of the house"
          type="number"
        />
        <span
          class="error-note"
          data-testid="form__error"
        />
      </label>
      <label
        class="small-field"
      >
        <h3>
          Price:
        </h3>
        <div
          class="price"
        >
          <input
            data-testid="form__price"
            name="price"
            placeholder="Enter price"
            type="number"
          />
          <select
            name="currency"
          >
            <option
              selected=""
              value="$"
            >
              $
            </option>
            <option
              value="€"
            >
              €
            </option>
            <option
              value="₽"
            >
              ₽
            </option>
          </select>
        </div>
        <span
          class="error-note"
          data-testid="form__price-error"
          id="price-error"
        />
      </label>
      <label
        class="small-field"
      >
        <h3>
          Date of construction:
        </h3>
        <input
          data-testid="form__date"
          name="date"
          type="date"
        />
        <span
          class="error-note"
          data-testid="form__date-error"
        />
      </label>
      <div
        class="middle-field ad-type"
      >
        <h3>
          Type of ad:
        </h3>
        <input
          checked=""
          id="sale"
          name="typeAdd"
          type="radio"
          value="sale"
        />
        <label
          class="switcher"
          for="sale"
        >
          <span>
            Sale
          </span>
        </label>
        <input
          id="rent"
          name="typeAdd"
          type="radio"
          value="rent"
        />
        <label
          class="switcher"
          for="rent"
        >
          <span>
            Rent
          </span>
        </label>
      </div>
      <label
        class="middle-field"
      >
        <h3>
          Ready for use
        </h3>
        <input
          name="ready"
          type="checkbox"
        />
      </label>
      <label
        class="big-field"
      >
        <h3>
          Add images:
        </h3>
        <input
          data-testid="form__file"
          name="img"
          type="file"
        />
        <span
          class="error-note"
          data-testid="form__file-error"
        />
      </label>
      <input
        class="submit-button"
        data-testid="form__submit"
        disabled=""
        type="submit"
        value="send"
      />
    </form>
    <div
      class="ads-list"
      data-testid="Cardlist"
    />
  </div>
</div>
`;
